lib/pbench/test/unit/server/query_apis/test_query_builder.py

```diff
@@ -10,114 +12,110 @@ class TestQueryBuilder:
     def elasticbase(self, client):
         return ElasticBase(client.config, client.logger, Schema())
 
-    @pytest.fixture()
-    def current_user_drb(self, monkeypatch):
-        class FakeHTTPTokenAuth:
-            def current_user(self) -> User:
-                return User(
-                    email="email@example.com",
-                    id=3,
-                    username="drb",
-                    first_name="Test",
-                    last_name="Account",
-                )
-
-        with monkeypatch.context() as m:
-            m.setattr(Auth, "token_auth", FakeHTTPTokenAuth())
-            yield
-
-    @pytest.fixture()
-    def current_user_none(self, monkeypatch):
-        class FakeHTTPTokenAuth:
-            def current_user(self) -> User:
-                return None
-
-        with monkeypatch.context() as m:
-            m.setattr(Auth, "token_auth", FakeHTTPTokenAuth())
-            yield
-
     @pytest.fixture()
     def current_user_admin(self, monkeypatch):
+        admin_user = User(
+            email="email@example.com",
+            id=6,
+            username="admin",
+            first_name="Test",
+            last_name="Admin",
+            role="admin",
+        )
+
         class FakeHTTPTokenAuth:
             def current_user(self) -> User:
-                return User(
-                    email="email@example.com",
-                    id=6,
-                    username="admin",
-                    first_name="Test",
-                    last_name="Admin",
-                    role="admin",
-                )
+                return admin_user
 
         with monkeypatch.context() as m:
             m.setattr(Auth, "token_auth", FakeHTTPTokenAuth())
-            yield
+            yield admin_user
 
-    def test_user_and_access(
-        self, elasticbase, server_config, current_user_drb, user_ok
-    ):
+    def test_user_and_access(self, elasticbase, server_config, current_user_drb):
         """
         Test the query builder when both user and access are specified. This is
         a simple path; the builder assumes that there is an authenticated user
         and that user has access to the specified dataset owner. We just build
         both terms into the query.
         """
+        id = str(current_user_drb.id)
         query: JSON = elasticbase._get_user_query(
-            {"user": "drb", "access": "public"}, [{"term": {"icecream": "vanilla"}}]
+            {"user": id, "access": "public"}, [{"term": {"icecream": "vanilla"}}]
         )
         assert query == {
             "bool": {
                 "filter": [
                     {"term": {"icecream": "vanilla"}},
+                    {"term": {"authorization.owner": id}},
                     {"term": {"authorization.access": "public"}},
-                    {"term": {"authorization.owner": "drb"}},
                 ]
             }
         }
```

Comment:
I wonder if it would be worth sorting the clauses or otherwise making this test resilient against small changes in the CUT.